### PR TITLE
mail-client/balsa: version bump to 2.6.4

### PR DESCRIPTION
**Pull request by Schievel1**
Signed-off-by: Pascal Jaeger <pascal.jaeger@leimstift.de>

### CODE REVIEW

**Comment by mattst88** [comment]
Oops, didn't mean to push that. Reopening.

File: mail-client/balsa/balsa-2.6.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		$(use_with gnome gtksourceview)
+		--with-spell-checker=gspell
+		$(use_with ldap)
+		$(use_with rubrica)
+		--without-osmo
+		$(use_with sqlite)
+		$(use_with gnome-keyring libsecret)
+		--without-gcr # experimental
+	)
+	gnome2_src_configure "${myconf[@]}"
```
Comment (mattst88) [inline mail-client/balsa/balsa-2.6.4.ebuild:79]: Would be good to switch to meson, if possible.

**Comment by Schievel1** [comment]
I can not get this to build anyway. 

With old build system it comes up with 
```
O1 -Wl,--as-needed -Wsign-compare -Wpointer-arith -Wnested-externs -Wchar-subscripts -Wuninitialized -Wmissing-prototypes -Wreturn-type -Wdeclaration-after-statement -Wredundant-decls -Wshadow -Wformat -Wformat-security -c -o gmime-gpgme-signature.o gmime-gpgme-signature.c
body.c: In function ‘body_weak_notify’:
body.c:74:29: error: ‘LibBalsaMessageBody’ {aka ‘struct _LibBalsaMessageBody’} has no member named ‘selection_table’
   74 |     g_hash_table_remove(body->selection_table, key);
      |                             ^~
body.c: In function ‘libbalsa_message_body_free’:
body.c:114:13: error: ‘LibBalsaMessageBody’ {aka ‘struct _LibBalsaMessageBody’} has no member named ‘selection_table’
  114 |     if (body->selection_table != NULL) {
      |             ^~
body.c:115:34: error: ‘LibBalsaMessageBody’ {aka ‘struct _LibBalsaMessageBody’} has no member named ‘selection_table’
  115 |         g_hash_table_foreach(body->selection_table, selection_table_foreach, body);
      |                                  ^~
body.c:116:34: error: ‘LibBalsaMessageBody’ {aka ‘struct _LibBalsaMessageBody’} has no member named ‘selection_table’
  116 |         g_hash_table_destroy(body->selection_table);
      |                                  ^~
make[3]: *** [Makefile:837: body.o] Error 1
```

with meson it says 
```
Running custom install script '/usr/bin/gtk-update-icon-cache --ignore-theme-index /usr/share/balsa'
--- stdout ---

--- stderr ---
gtk-update-icon-cache: Failed to open file /usr/share/balsa/.icon-theme.cache : No such file or directory

FAILED: install script '/usr/bin/gtk-update-icon-cache --ignore-theme-index /usr/share/balsa' exit code 1, stopped
```

The latter could be something with gentoos way of building, idk. I will file an issue upstream.

**Comment by mattst88** [comment]
Yeah, I saw the same build error with autotools.

Can you update the PR with your meson version?

**Comment by Schievel1** [comment]
It's in there as comment. Just need to inherit meson instead of gnom2

**Comment by Schievel1** [comment]
I updated the build to meson and removed the call of gtk-update-icon-cache, since we do that ourselves anyway in pkg_postinsall we don't need it.

File: mail-client/balsa/balsa-2.6.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+# DOCS="AUTHORS ChangeLog HACKING NEWS README TODO docs/*"
+
+src_prepare() {
+	default
+	sed -i 's/if gtk_update_icon_cache_program.found()/if false/' images/meson.build || die 
+}
+
+src_configure() {
+	local myconf=(
```
Comment (mattst88) [inline mail-client/balsa/balsa-2.6.4.ebuild:66]: `meson_src_configure` looks for `emesonargs`, so none of this is working.

File: mail-client/balsa/balsa-2.6.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		-Dgcr=false
+		-Dmore-warnings=true
+		-Dhelp-files=false
+		-Dlibnetclient-docs=false
+		-Dlibnetclient-test=false
+	)
+	if use webkit; then
+		myconf+=" -Dhtml-widget=webkit2"
+	else
+		mycond+=" -Dhtml-widget=no"
```
Comment (mattst88) [inline mail-client/balsa/balsa-2.6.4.ebuild:92]: Need to fix these too (and this one is typo'd).

File: mail-client/balsa/balsa-2.6.4.ebuild (hunk 2 above)
Comment (Schievel1) [reply mail-client/balsa/balsa-2.6.4.ebuild]: Oh that is the reason why the build worked in meson: I never passed the options to it. WIthout that it uses -Dhtml-widged=no as the default, which causes the build to fail. 
I corrected the emesonargs and pushed a version that sets -Dhtml-widget=webkit2 unconditionally.

**Pull request closed without merging**